[WebVTT] Cues should use white-space:pre-line

jernoble opened the pull request:
#### 36f38359156a0215fa0147c38f6c2e62e745bd89
<pre>
[WebVTT] Cues should use white-space:pre-line
<a href="https://bugs.webkit.org/show_bug.cgi?id=275616">https://bugs.webkit.org/show_bug.cgi?id=275616</a>
<a href="https://rdar.apple.com/130084169">rdar://130084169</a>

Reviewed by Eric Carlson.

The WebVTT spec requires:

&gt; 7.4. Applying CSS properties to WebVTT Node Objects:
&gt; The white-space property on the (root) list of WebVTT Node Objects must be set to pre-line. [CSS22]

Update our default styles for cues to use this value for `white-space:`. Add WPT reference tests to
verify the change. The test uses `mix-blend-mode` to combine the actual rendering and the expected
rendering into a green result.

* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/README.md: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/basic-cue-rendering-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/basic-cue-rendering.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/basic-cue-rendering.webvtt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/content/silence.m4a: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/content/silence.mp3: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/reference.css: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/white-space-pre-line-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/white-space-pre-line.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/white-space-pre-line.webvtt: Added.
* Source/WebCore/Modules/modern-media-controls/controls/text-tracks.css:
(::-webkit-media-text-track-display):
* Source/WebCore/css/mediaControls.css:
(video::-webkit-media-text-track-display):

Canonical link: <a href="https://commits.webkit.org/280554@main">https://commits.webkit.org/280554@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3b85aa2f0a88d4cfa963a11c211c1dabc0424b77

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56939 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/36267 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9413 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60559 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7382 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/59067 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43890 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7572 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/46117 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5185 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58969 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/34061 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/49139 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26975 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30841 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/6470 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6387 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/52801 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6741 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/62240 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/852 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6848 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/53372 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/855 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/49194 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/53414 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12592 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/723 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/32096 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/33181 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/34266 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32927 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

danae404 commented on Source/WebCore/css/mediaControls.css:
```diff
@@ -263,7 +263,7 @@ video::-webkit-media-text-track-region-container.scrolling {
     position: absolute;
     overflow: hidden;
-    white-space: pre-wrap;
+    white-space: pre-line;
```
Because this rule is more specific than the one in text-tracks.css, is it necessary to have?

Ahmad-S792 replied:
FYI - `mediaControls.css` is only used by Sony / Playstation port and I have highlighted it to @fujii that it would be good to explore `text-tracks.css` as single source so we can do this clean-up and remove this one. (Just mentioned and didn't discussed any future plans).

jernoble committed:
[WebVTT] Cues should use white-space:pre-line
https://bugs.webkit.org/show_bug.cgi?id=275616
rdar://130084169

Reviewed by Eric Carlson.

The WebVTT spec requires:

> 7.4. Applying CSS properties to WebVTT Node Objects:
> The white-space property on the (root) list of WebVTT Node Objects must be set to pre-line. [CSS22]

Update our default styles for cues to use this value for `white-space:`. Add WPT reference tests to
verify the change. The test uses `mix-blend-mode` to combine the actual rendering and the expected
rendering into a green result.

* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/README.md: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/basic-cue-rendering-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/basic-cue-rendering.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/basic-cue-rendering.webvtt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/content/silence.m4a: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/content/silence.mp3: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/reference.css: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/white-space-pre-line-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/white-space-pre-line.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webvtt/rendering/white-space-pre-line.webvtt: Added.
* Source/WebCore/Modules/modern-media-controls/controls/text-tracks.css:
(::-webkit-media-text-track-display):
* Source/WebCore/css/mediaControls.css:
(video::-webkit-media-text-track-display):

Canonical link: https://commits.webkit.org/280554@main
